Add implemention of the Ballerina Connector for Azure Cosmos DB (#2)
ballerina-platform/module-ballerinax-azure-cosmosdb: Module.md

> The Azure Cosmos DB is Microsoft’s NOSQL database in Azure technology stack. It is called a globally distributed 

Reviewer: ```suggestion
The Azure Cosmos DB is Microsoft’s NoSQL database in the Azure technology stack. It is called a globally distributed 
```

Author: Resolved this thanks

> multi-model database which is used for managing data across the world. Key purposes of the Azure CosmosDB is to achieve 

Reviewer: ```suggestion
multi-model database which is used for managing data across the world. The key purpose of the Azure CosmosDB is to achieve 
```

low latency and high availability while maintaining a flexible scalability. 

> The Ballerina Cosmos DB connector allows you to connect to a Azure Cosmos DB resource from Ballerina and perform various 

Reviewer: ```suggestion
The Ballerina Cosmos DB connector allows you to connect to an Azure Cosmos DB resource from Ballerina and perform various 
```

operations such as `find`, `create`, `read`, `update`, and `delete` operations of `Databases`, `Containers`,
`User Defined Functions`, `Tiggers`, `Stored Procedures`, `Users`, `Permissions` and `Offers`. 

## Compatibility

|                           |    Version                  |
|:-------------------------:|:---------------------------:|
| Ballerina Language        | Swan-Lake-Preview8          |
| Cosmos DB API Version     | 2018-12-31                  |

## CosmosDB Clients

> There are two clients provided by Ballerina to interact with CosmosDB.

Reviewer: ```suggestion
There are two clients provided by Ballerina to interact with Cosmos DB.
```

Reviewer: Check all the places

Author: Checked and changed Module.md, Package.md, README.md and Utils.bal

1. **cosmosdb:DataPlaneClient** - This connects to the running CosmosDB databases and containers to execute data-plane 
operations 

   ```ballerina
    cosmosdb:Configuration configuration = {
        baseUrl : <BASE_URL>,
        masterOrResourceToken : <MASTER_OR_RESOURCE_TOKEN>,
    };
    cosmosdb:DataPlaneClient coreClient = new(configuration);
   ```
2. **cosmosdb:ManagementClient** - This connects to the running CosmosDB databases and containers to execute 
management-plane operations 

   ```ballerina
    cosmosdb:Configuration configuration = {
        baseUrl : <BASE_URL>,
        masterOrResourceToken : <MASTER_OR_RESOURCE_TOKEN>,
    };
    cosmosdb:ManagementClient managementClient = new(configuration);
   ```

## Samples 
### Creating a database
For creating a database in Azure we have to provide a unique database ID which does not already exist in the specific 

> cosmos DB account. This operation will return a record of type Database. This will contain the success as true if the 

Reviewer: ```suggestion
Cosmos DB account. This operation will return a record of type Database. This will contain the success as true if the 
```

Author: Resolved and applied changes

operation is successful.

```ballerina
import ballerina/log;
import ballerinax/azure_cosmosdb as cosmosdb;

public function main() {
    cosmosdb:Configuration configuration = {
        baseUrl : "https://cosmosconnector.documents.azure.com:443",
        masterOrResourceToken : "mytokenABCD==",
    };
    cosmosdb:ManagementClient managementClient = new(configuration);

    cosmosdb:Database databaseResult = checkpanic managementClient->createDatabase(<DATABASE_ID>);
}
```

### Creating a Container

> A container can be created inside an existing database in the cosmos DB account. As the REST api version which is used 

Reviewer: ```suggestion
A container can be created inside an existing database in the cosmos DB account. As the REST API version which is used 
```

Author: Resolved

in this implementation of the connector strictly supports the partition key, it is a necessity to provide the 
partition key in the creation of a container. 

```ballerina
import ballerina/log;

> import ballerinax/azure_cosmosdb as cosmosdb;

Reviewer: Why the module name is `azure_cosmosdb`? Shouldn't it be `azure.cosmosdb`?

Author: For now the connector is in SLP8. So, it doesn't support adding dots in the module name. Once the connector migrate to slalpha1 this name will be changed as `azure.cosmosdb`

Reviewer: ack.

public function main() {
    cosmosdb:Configuration configuration = {
        baseUrl : "https://cosmosconnector.documents.azure.com:443",
        masterOrResourceToken : "mytokenABCD==",
    };
    cosmosdb:ManagementClient managementClient = new(configuration);

    cosmosdb:PartitionKey partitionKey = {
        paths: ["/accountNumber"],
        kind :"Hash",
        'version: 2
    };
    cosmosdb:Container containerResult = checkpanic managementClient->createContainer(<DATABASE_ID>, <CONTAINER_ID>, 
            partitionKey);
}
```

> ### Inserting a Doument

Reviewer: ```suggestion
### Inserting a Document
```

> Azure cosmos DB allows the execution of CRUD operations on items separately. As we are using the Core API underneath 

Reviewer: ```suggestion
Azure Cosmos DB allows the execution of CRUD operations on items separately. As we are using the Core API underneath 
```

Author: Resolved

the connector, an item may refer to a document in the container. SQL API stores entities in JSON in a hierarchical 
key-value document. The max document size in Cosmos DB is 2 MB.

```ballerina
import ballerina/log;
import ballerinax/azure_cosmosdb as cosmosdb;

public function main() {
    cosmosdb:Configuration configuration = {
        baseUrl : "https://cosmosconnector.documents.azure.com:443",
        masterOrResourceToken : "mytokenABCD==",
    };
    cosmosdb:DataPlaneClient coreClient = new (configuration);

    record {|string id; json...;|} documentBody = {
        id: "documentid1",
        LastName: "Sheldon", 
        accountNumber: 001234222
    };

    cosmosdb:Document documentResult1 = checkpanic coreClient->createDocument(<DATABASE_ID>, <CONTAINER_ID>, documentBody, 
            <VALUE_OF_PARTITIONKEY>); 
}
```
### List Documents

> Usually Cosmos DB provides an json array of json objects as the response for list operations. But, the connector 

Reviewer: ```suggestion
Usually, Cosmos DB provides an array of JSON objects as the response for list operations. But, the connector 
```

Author: Resolved
